File: Javascript/ConstructStringFromBinaryTree.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0:val
    this.left = (left === undefined) ? null:left
    this.right = (right === undefined) ? null:right 
}


var stringFromBinaryTree = function(root){
    let outputString = ""

    function helper(node){
        if(node === null){
            return
        }

        outputString += node.val.toString()

        if(node.left === null && node.right === null){
            return
        }
        else{
            outputString += "("
            helper(node.left)
            outputString += ")"
        }

        if(node.right){
            outputString += "("
            helper(node.right)
            outputString += ")"
        }

    }

    helper(root)
    return outputString
}

File: Python/ConstructStringFromBinaryTree.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left   
        self.right = right

class Solution:
    def stringFromBinaryTree(root):
        outputString = ""

        def helper(node,outputString):
            if not node:
                return outputString
            
            outputString += str(node.val)

            if not node.left and not node.right:
                return outputString
            else:
                outputString += "("
                outputString = helper(node.left,outputString)
                outputString += ")"

            if node.right:
                outputString += "("
                outputString = helper(node.right,outputString)
                outputString += ")"

            return outputString
        
        outputString = helper(root,outputString)

        return outputString